### PR TITLE
[bugfix] Fix `--restore-session` failure when ReFrame root directory is not in `sys.path[0]`

### LINKED ISSUE #2135: Errors when ReFrame root directory is not first path in `sys.path`
Hello,

The context here is when I use ReFrame CI integration capabilities, I am getting errors in the dependent jobs. When we use `--restore-session` flag for dependent job, ReFrame is detecting modules from filenames [here](https://github.com/eth-cscs/reframe/blob/bb779ce73a4f081aef1b6e0c20c127061bec24d1/reframe/utility/__init__.py#L84). 

I am not sure of the rabbit hole details of why it is being done, but the issue I have noticed is that on that stated line, we implicitly assume `sys.path[0]` if ReFrame root directory. If it is not, ReFrame will have problems finding the modules. For example, in this [failed CI pipeline](https://gitlab.com/ska-telescope/sdp/ska-sdp-benchmark-tests/-/jobs/1512352442#L87), I printed the `sys.path` and we can see that ReFrame root directory is not the first path.

This is the directory structure of my repository:
```
.
├── apps
├── config
├── docs
├── helloworld
├── modules
├── output
├── perflogs
├── reframe
├── spack
└── topologies
```

And I usually launch tests from root of the repo using:
`reframe/bin/reframe -C <config> -c <test> -r`

### PR DESCRIPTION
**Pull request by rsarm**
Close #2135

### CODE REVIEW

**Comment by rsarm** [comment]
@mahendrapaipur could you have a look?

**Comment by mahendrapaipuri** [comment]
@rsarm Looks good. Just a minor suggestion. Do you think moving 
```
rfm_root = os.path.realpath(os.path.join(os.path.dirname(__file__), '../..'))
```
into a new function like `get_rfm_root` makes sense? This makes it more modular and reusable if we ever need ReFrame root directory in further developments.

**Review by teojgo — APPROVED** [review]

File: reframe/utility/__init__.py [hunk 1]
```diff
@@ -81,7 +81,9 @@ def import_module_from_file(filename, force=False):
         filename = os.path.join(filename, '__init__.py')
 
     # Express filename relative to reframe
-    rel_filename = os.path.relpath(filename, sys.path[0])
+    rfm_root = os.path.realpath(os.path.join(os.path.dirname(__file__),
+                                '../..'))
```
Comment (vkarak) [inline reframe/utility/__init__.py:85]: Isn't that equivalent to this?

```suggestion
    rfm_root = os.path.realpath(reframe.INSTALL_PREFIX)
```

And another question is that it's not obvious to me the purpose of `os.path.realpath` here.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: reframe/utility/__init__.py (hunk 1 above)
Comment (rsarm) [reply reframe/utility/__init__.py:85]: The `os.path.realpath` was to get the resulting path without the `../../`

**Comment by rsarm** [comment]
What about replacing by [here](https://github.com/eth-cscs/reframe/blob/9804e1cace17229a656ad7cb9b28069cee2437cf/reframe/core/exceptions.py#L308 ) `reframe.INSTALL_PREFIX` too?

**Review by vkarak — APPROVED** [review]

**Pull request merged**